hooks/gen_docs/__init__.py

```diff
@@ -2,25 +2,23 @@
 
 from collections.abc import Generator
 from enum import Enum
-from typing import Any, Generic, TypeVar
+from typing import Any
 
-_T = TypeVar("_T")
 
-
-class SuperEnum(Generic[_T], Enum):
+class StrEnum(str, Enum):
     """Adds constructors that return all items in a ``Generator``.
 
     Introduces constructors that return a ``Generator`` object
     either containing all items, only their names or their values.
     """
 
     @classmethod
-    def items(cls) -> Generator[tuple[_T, Any], None, None]:
+    def items(cls) -> Generator[tuple[Any, str], None, None]:
         """Return all item names and values in tuples."""
         return ((e.name, e.value) for e in cls)
 
     @classmethod
-    def keys(cls) -> Generator[_T, None, None]:
+    def keys(cls) -> Generator[str, None, None]:
```

Comment:
consider annotating as `Iterator[str]`

Reply:
Could I annotate the other 2 as `Iterator`, too?

Reply:
yep